docker/Dockerfile

```
@@ -145,8 +145,15 @@ WORKDIR /source
 
 # Install (non-development) Python dependencies of Nautobot
 RUN --mount=type=cache,target="/root/.cache",sharing=locked \
-    poetry install --no-root --only main --no-ansi --extras all && \
-    rm -rf /tmp/tmp*
+    if poetry install --no-root --only main --no-ansi --extras all; then \
+        echo "Parallel installation succeeded"; \
+    else \
```

Note that `POETRY_INSTALLER_PARALLEL` is a build arg, so maybe for most correctness this should be something like:

```suggestion
        echo "Installation succeeded"; \
    elif [ ${POETRY_INSTALLER_PARALLEL} == "true" ]; then \
```

---

hadolint was complaining lmk if this change looks reasonable https://github.com/nautobot/nautobot/pull/6693/commits/6cfad49b812f2577241af86c07297d03f13f80b8